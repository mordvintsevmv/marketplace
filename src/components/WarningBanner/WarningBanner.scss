.in-progress{
  position: relative;

  width: 30vw;
  min-width: 300px;

  min-height: 60px;
  margin: 0 auto 40px auto;

  border: 1px solid #E98585;
  border-radius: 25px;

  background-color: rgba(233,133,133,.05);

  padding: 15px;

  &__top{
    display: flex;

    gap: 15px;
    align-items: center;
    justify-content: flex-start;
    margin-bottom: 25px;
  }

}