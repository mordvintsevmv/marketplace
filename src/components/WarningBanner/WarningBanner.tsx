import {FC} from "react";
import {Warning} from "@mui/icons-material";
import "./WarningBanner.scss"

interface InProgressProps {
    title: string;
    description: string;
}

const WarningBanner: FC<InProgressProps> = ({title, description}) => {
    return (
        <div className={"in-progress"}>

            <div className={"in-progress__top"}>
                <div className={"in-progress__warn"}>
                    <Warning/>
                </div>

                <div className={"in-progress__title"}>
                    <h4>{title}</h4>
                </div>

            </div>

            <div className={"in-progress__description"}>
                {description}
            </div>
        </div>
    )
}

export default WarningBanner