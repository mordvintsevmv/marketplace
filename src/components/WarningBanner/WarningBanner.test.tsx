import {render, screen} from '@testing-library/react'
import WarningBanner from "./WarningBanner";

const title: string = 'Test title'
const description: string = 'Test description'

describe('Testing WarningBanner Component', () => {
    test('Component render', () => {
        render(<WarningBanner title={title} description={description}/>)

        expect(screen.getByText('Test title')).toBeInTheDocument();
        expect(screen.getByText('Test description')).toBeInTheDocument();
    })

    test('WarningBanner snapshot', ()=>{
        const view = render(<WarningBanner title={title} description={description}/>);

        expect(view).toMatchSnapshot()
    })
})