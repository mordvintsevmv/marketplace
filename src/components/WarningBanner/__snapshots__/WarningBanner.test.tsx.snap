// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing WarningBanner Component WarningBanner snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="warning-banner"
      >
        <div
          class="warning-banner__top"
        >
          <div
            class="warning-banner__warn"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="WarningIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
              />
            </svg>
          </div>
          <div
            class="warning-banner__title"
          >
            <h4>
              Test title
            </h4>
          </div>
        </div>
        <div
          class="warning-banner__description"
        >
          Test description
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="warning-banner"
    >
      <div
        class="warning-banner__top"
      >
        <div
          class="warning-banner__warn"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="WarningIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
        </div>
        <div
          class="warning-banner__title"
        >
          <h4>
            Test title
          </h4>
        </div>
      </div>
      <div
        class="warning-banner__description"
      >
        Test description
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
